mods/mixandmega/scripts.js

```diff
@@ -90,10 +90,12 @@ let BattleScripts = {
 		if (megaTemplate.isPrimal) deltas.isPrimal = true;
 		return deltas;
 	},
-	doGetMixedTemplate: function (template, deltas) {
+	doGetMixedTemplate(templateOrTemplateName, deltas) {
 		if (!deltas) throw new TypeError("Must specify deltas!");
-		if (!template || typeof template === 'string') template = this.getTemplate(template);
-		template = Object.assign({}, template);
+		if (!templateOrTemplateName || typeof templateOrTemplateName === 'string') {
+			templateOrTemplateName = this.getTemplate(templateOrTemplateName);
+		}
+		let template = /** @type {any} */ (Object.assign({}, templateOrTemplateName));
```

Comment:
Missed a `deepClone` here?
Also, I don't think the call to `this.getTemplate` needs to be guarded like that.

Reply:
Good catch.

What do you mean by guarded?

Reply:
...that's a good point.